static/app/views/automations/components/ruleNodeList.tsx

```diff
@@ -11,27 +11,28 @@ import RuleNode from 'sentry/views/automations/components/ruleNode';
 
 interface RuleNodeListProps {
   conditions: NewDataCondition[];
+  dataConditionTypes: DataConditionType[];
   group: string;
   onAddRow: (type: DataConditionType) => void;
   onDeleteRow: (id: number) => void;
   placeholder: string;
   updateCondition: (index: number, condition: Record<string, any>) => void;
 }
 
-// TODO: only show data conditions that are returned by the API
-const options = Array.from(dataConditionNodesMap, ([key, value]) => ({
-  value: key,
-  label: value.label,
-}));
-
 export default function RuleNodeList({
+  dataConditionTypes,
   group,
   placeholder,
   conditions,
   onAddRow,
   onDeleteRow,
   updateCondition,
 }: RuleNodeListProps) {
+  const options = Array.from(dataConditionNodesMap, ([key, value]) => ({
+    value: key,
+    label: value.label,
+  })).filter(option => dataConditionTypes.includes(option.value));
```

Comment:
Might be a bit cleaner
```suggestion
  const options = dataConditionNodesMap.entries().map([value, { label }]) => ({ value, label })).filter(({ value }) => dataConditionTypes.includes(value));
```

Reply:
fixed in https://github.com/getsentry/sentry/pull/89928/commits/cc1ce4f42355743dcd3194ad87ce4eb936ea609b

i still had to do `Array.from()` to not get errors (i'm still not familiar with typescript 😭 )